### PR TITLE
fix: Enable pytype on Pub/Sub Lite repo and fix all errors

### PR DESCRIPTION
**Pull request by dpcollins-google**
Workarounds for https://github.com/google/pytype/issues/977 and https://github.com/google/pytype/issues/978 are present in this PR, as well as a backport of https://github.com/googleapis/gapic-generator-python/pull/970

Also modifies construction of CreateSubscriptionRequests slightly

### CODE REVIEW

File: google/cloud/pubsublite/internal/wire/admin_client_impl.py [hunk 1]
```diff
@@ -88,12 +89,12 @@ def create_subscription(
     ) -> Subscription:
         path = SubscriptionPath.parse(subscription.name)
         return self._underlying.create_subscription(
-            request={
-                "parent": str(path.to_location_path()),
-                "subscription": subscription,
-                "subscription_id": path.name,
-                "skip_backlog": (starting_offset == BacklogLocation.END),
-            }
+            request=CreateSubscriptionRequest(
```
Comment (tswast) [inline google/cloud/pubsublite/internal/wire/admin_client_impl.py:92]: This is potentially a breaking behavior change due to some known issues in proto-plus if you have any users that have passed in a subscription as a dictionary

https://github.com/googleapis/proto-plus-python/issues/161 comes to mind, but I think there was another one specifically about mixing dictionaries and protos, but I'm having trouble finding it.

Comment (tswast) [reply google/cloud/pubsublite/internal/wire/admin_client_impl.py:92]: Found it! https://github.com/googleapis/python-tasks/issues/148

Comment (dpcollins-google) [reply google/cloud/pubsublite/internal/wire/admin_client_impl.py:92]: Acknowledged. I'm okay with this- the type hints tell the user the exact type they need to pass. I don't think this library has wide enough usage that its expected to be bug-compatable with past versions: this is only a breaking change if proto-plus-python has a bug preventing usage of a dict where it should be usable, and the user is ignoring the type hint provided on the method.

File: google/cloud/pubsublite/cloudpubsub/publisher_client_interface.py [hunk 2]
```diff
@@ -50,9 +50,10 @@ async def publish(
         Raises:
           GoogleApiCallError: On a permanent failure.
         """
+        raise NotImplementedError()
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply google/cloud/pubsublite/cloudpubsub/publisher_client_interface.py:53]: I'm going to actually purge the error messages- this seems like a QOI thing, I think NotImplementedError being raised from a code location in an interface class with @abstractmethod annotated on it should be sufficient to let the user know they have an instance of the interface not the implementation.

File: google/cloud/pubsublite/cloudpubsub/internal/single_publisher.py [hunk 3]
```diff
@@ -70,3 +71,4 @@ def publish(
         Raises:
           GoogleApiCallError: On a permanent failure.
         """
+        raise NotImplementedError("")
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply google/cloud/pubsublite/cloudpubsub/internal/single_publisher.py]: Removed, discussed in other comment my rationale.

File: google/cloud/pubsublite/internal/wire/admin_client_impl.py (hunk 1 above)
Comment (meredithslota) [reply google/cloud/pubsublite/internal/wire/admin_client_impl.py:92]: I haven't looked closely enough to know whether it's a breaking change or not, but if it is, we should update the issue title to not use `chore` accordingly.

Comment (dpcollins-google) [reply google/cloud/pubsublite/internal/wire/admin_client_impl.py:92]: Done.

**Review by manuelmenzella-google — APPROVED** [review]

**Pull request merged**